### ISSUE TITLE
[4.2] Smart search indexer button does not always work in multilingual site

### ISSUE THREAD

**Issue by crystalenka** [joomla/joomla-cms#39717]
### Steps to reproduce the issue
1. Set up a blank Joomla installation and install an additional language of your choice.
2. Install multilingual sample data.
3. Go to Smart Search > Index > and then click the "Index" button.
4. If it works as expected, go to options, change something in the index settings, and then try again.

### Expected result
Site is indexed.

### Actual result
Frozen index modal which never errors out or succeeds. (I've left it for something like 15-20 minutes just in case.)

<img width="815" alt="Screenshot 2023-01-26 at 11 59 50" src="https://user-images.githubusercontent.com/9141288/214808019-cefd3832-74d2-465e-93cc-ce9e14849529.png">

Not sure if the background indexing is okay and the modal just gets stuck.

Possibly related console error:
```
[Error] SyntaxError: JSON Parse error: Unrecognized token '<'
	parse (indexer.min.js:1:2366)
	onSuccess (indexer.min.js:1:2366)
	(anonymous function) (core.min.js:1:6230)
```

Possibly related PHP notice:

```
PHP Deprecated:  Locale::getPrimaryLanguage(): Passing null to parameter #1 ($locale) of type string is deprecated in /administrator/components/com_finder/src/Indexer/Helper.php on line 357
```

### System information (as much as possible)
PHP 8.1
Joomla 4.2.6 and 4.2.7-RC2

### Additional comments
Doesn't always happen, I haven't been able to pin down the conditions in which it happens vs it doesn't.

**Comment by brianteeman** [comment]
have you tried "enable logging" in the finder component options. Might give some clues.

**Comment by crystalenka** [comment]
I've done that now and tried again, but not sure where to find the log.

**Comment by brianteeman** [comment]
No idea either - I would assume administrator/logs

You also get some more detailed info if you have debug on when running the index iirc

**Comment by crystalenka** [comment]
Found it.

This is what's logged within a few seconds after I clear the index and start from scratch:

```
2023-01-26T10:19:20+00:00	INFO ::1	-	Starting the indexer
2023-01-26T10:19:20+00:00	INFO ::1	-	Starting the indexer batch process
2023-01-26T10:19:20+00:00	INFO ::1	-	Batch completed, peak memory usage: 6,291,456 bytes
```
So it appears that the index works fine, but something funky is happening with the modal as it's still frozen like in my screenshot above.

> You also get some more detailed info if you have debug on when running the index iirc

Interesting! Will try that next.

**Comment by crystalenka** [comment]
Modal is still frozen with debug on, with additional frozen debug info haha:

<img width="815" alt="Screenshot 2023-01-26 at 12 22 23" src="https://user-images.githubusercontent.com/9141288/214812743-53a344b7-d543-48cc-abd3-3625b5832a8f.png">

I'm going to guess this is some kind of javascript error based on the error in the console.

**Comment by wilsonge** [comment]
From your error it's more likely the php warning to be honest. If you look in the network tab of dev tools

![image](https://user-images.githubusercontent.com/1986000/215355949-ac00077c-076a-4428-876d-17768cd6799b.png)

Filter by XHR requests to make life easier you should be able to see the raw response coming back from the server and which one isn't valid JSON. Sorry I've tried reproducing on my PHP 8.1 setup and can't seem to reproduce with error reporting on maximum :( otherwise would do it myself!

**Comment by Hackwar** [comment]
@crystalenka Did you check this again?

**Comment by crystalenka** [comment]
@Hackwar it's on my list for today :)

**Comment by Hackwar** [comment]
We have a fix in #39965. Closing this one. :-)

**Comment by wilsonge** [comment]
This stops the output from hard crashing but doesn't solve @crystalenka 's underlying root cause of some sort of php error so would still be good to get that info please Crystal!
